packages/snap-store-mobx/src/Finder/FinderStore.ts

```diff
@@ -40,11 +52,70 @@ export class FinderStore extends AbstractStore {
 		}
 	}
 
-	update(data: SearchResponseModel & { meta: MetaResponseModel }): void {
+	save(): void {
+		if (this.config.persist?.enabled && this.selections.filter((selection) => selection.selected).length) {
+			this.persistedStorage.set('config', this.config);
+			this.persistedStorage.set('data', this.data);
+			this.persistedStorage.set('date', Date.now());
+			this.persistedStorage.set(
+				'selections',
+				this.selections.map((selection) => {
+					return {
+						selected: selection.selected,
+						data: selection.data,
+						facet: selection.facet,
+					} as SelectedSelection;
+				})
+			);
+		}
+	}
+
+	reset = (): void => {
+		if (this.config.persist?.enabled) {
+			this.persistedStorage.clear();
+			this.state.persisted = false;
+		}
+
+		if (this.services.urlManager.state.filter) {
+			this.storage.clear();
+			this.selections = [];
+			this.loaded = false;
+		}
+	};
+
+	loadPersisted(): void {
+		if (this.config.persist?.enabled && !this.loaded) {
+			const date = this.persistedStorage.get('date');
+			const data = this.persistedStorage.get('data');
```

Comment:
Would this data always be equal to the last selectonStore data?
If so, could we possibly do away with `this.data` altogether?

Reply:
I think we should leave this because data contains information such as pagination data which you may always be rendering even if selections were persisted.

Reply:
Sounds good!